src/neptune/types/series/float_series.py

```diff
@@ -96,3 +103,22 @@ def unit(self):
 
     def __str__(self):
         return "FloatSeries({})".format(str(self.values))
+
+    def clear_of_unsupported_values(self):
```

Comment:
How about something more generic like:
```python
def filter_unsupported_values(values, steps, timestamps, filter_by):
    filtered = [
        (value, step, timestamp) for value, step, timestamp in zip(values, steps, timestamps) if filter_by(value)
    ]
    return (
        [value for value, _, _ in filtered],
        [step for _, step, _ in filtered],
        [timestamp for _, _, timestamp in filtered]
    )
```
and usage:
```python
        if steps is None:
            filled_steps = list(islice(cycle([None]), len(values)))
        else:
            assert len(values) == len(steps)
            filled_steps = steps

        if timestamps is None:
            filled_timestamps = list(islice(cycle([time.time()]), len(values)))
        else:
            assert len(values) == len(timestamps)
            filled_timestamps = timestamps

        clean_values, self._steps, self._timestamps = filter_unsupported_values(
            values=values,
            steps=filled_steps,
            timestamps=filled_timestamps,
            filter_by=is_unsupported_float
        )
        self._values = [float(value) for value in clean_values]
```

But it wasn't tested :D.

Reply:
That looks pretty! But we still need to warn user about unsupported, please take a look if my solution for that looks as pretty as this 😇